### PR TITLE
Doc Fixes

### PR DESCRIPTION
**Pull request by riedgar-ms**
A variety of fixes for the documentation:
- Get the base classes referenced correctly
- Add links to `sklearn` documentation from the `metrics`
- Expand `GridSearch` documentation

### CODE REVIEW

**Comment by riedgar-ms** [comment]
With a completely manual approach, I've managed to get this PR online:
https://fairlearn.github.io/fairlearn-doc-dev/
For some reason, the CSS files aren't loading, though

File: fairlearn/__init__.py [hunk 1]
```diff
@@ -1,3 +1,10 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""The fairlearn package contains a number of tools for
+analysing and mitigating disparity in Machine Learning models.
```
Comment (romanlutz) [inline fairlearn/__init__.py]: analyzing :-)

File: fairlearn/reductions/_grid_search/grid_search.py [hunk 2]
```diff
@@ -77,15 +77,36 @@ def accumulate_integer_grid(self, index, max_val):
...
     :type constraints: fairlearn.moments.Moment
+
+    :param selection_rule: Specifies the procedure for selecting the best model found by the
+        grid search. At the present time, the only valid value is "tradeoff_optimization" which
+        minimises a weighted sum of the error rate and constraint violation.
+    :type selection_rule: str
+
+    :param constraint_weight: When the `selection_rule` is "tradeoff_optimization" this specifies
+        the relative weight put on the constraint violation when selecting the best model.
+        The weight placed on the error rate will be `1-constraint_weight`
```
Comment (romanlutz) [inline fairlearn/reductions/_grid_search/grid_search.py]: what is 1-constraint_weight?

Comment (riedgar-ms) [reply fairlearn/reductions/_grid_search/grid_search.py]: That's supposed to be a mathematical statement - as in
`error_weight = 1 - constraint_weight`

File: fairlearn/__init__.py (hunk 1 above)
Comment (riedgar-ms) [reply fairlearn/__init__.py]: Mutter, mutter

**Pull request merged**